crates/compiler/mono/src/layout.rs

```diff
@@ -675,6 +675,7 @@ pub enum LayoutRepr<'a> {
     Builtin(Builtin<'a>),
     Struct(&'a [InLayout<'a>]),
     Boxed(InLayout<'a>),
+    Ptr(InLayout<'a>),
```

Comment:
Can we add some comments regarding the distinction between Boxed and Ptr? Namely that, if i understand correctly, Ptr is an alloca. I was thinking it would be too hard to do earlier, but maybe we could even reuse Ptr for pass-by-reference later.

Reply:
`Ptr` is just some pointer (heap or stack) without any reference counting. The (lack of) RC is the defining feature.